NAV_Algorithms/flight_observer.cpp

```diff
@@ -44,60 +44,90 @@ void flight_observer_t::update_every_10ms (
+
+      // horizontal kalman filter for velocity and acceleration in the air- (not ground) system
+      Kalman_v_a_observer_N.update ( gnss_velocity.e[NORTH] - wind_average.e[NORTH], ahrs_acceleration.e[NORTH]);
+      Kalman_v_a_observer_E.update ( gnss_velocity.e[EAST]  - wind_average.e[EAST],  ahrs_acceleration.e[EAST]);
+
+      // compute our kinetic energy in the air-system
+      specific_energy = (
+            SQR( gnss_velocity.e[NORTH] - wind_average.e[NORTH])
+	  + SQR( gnss_velocity.e[EAST]  - wind_average.e[EAST])
+	  + SQR( gnss_velocity.e[DOWN]) * vertical_energy_tuning_factor)
+	  * ONE_DIV_BY_GRAVITY_TIMES_2;
+
+      // speed-compensation type 1 = scalar product( air_velocity , acceleration) / g;
+      speed_compensation_INS_GNSS_1 = ahrs_instant_air_velocity * acceleration * RECIP_GRAVITY;
+
+      // INS-acceleration {scalar product *} air-velocity = speed compensation type 1
+      float3vector gnss_air_velocity = gnss_velocity - wind_average;
+      gnss_air_velocity.e[DOWN] = KalmanVario_GNSS.get_x ( KalmanVario_PVA_t::VARIO);
+
+      // speed compensation type 2 = air velocity * acceleration , both Kalman-filtered
+      speed_compensation_kalman_2 =
+	   (Kalman_v_a_observer_N.get_x ( Kalman_V_A_Aoff_observer_t::VELOCITY) * Kalman_v_a_observer_N.get_x ( Kalman_V_A_Aoff_observer_t::ACCELERATION)
+	  + Kalman_v_a_observer_E.get_x ( Kalman_V_A_Aoff_observer_t::VELOCITY) * Kalman_v_a_observer_E.get_x ( Kalman_V_A_Aoff_observer_t::ACCELERATION)
+	  + KalmanVario_GNSS.get_x (KalmanVario_PVA_t::VARIO) * KalmanVario_GNSS.get_x ( KalmanVario_PVA_t::ACCELERATION_OBSERVED)  * vertical_energy_tuning_factor)
+	  * RECIP_GRAVITY;
+
+      // speed compensation type 3 comes from the derivative of the specific energy
+      speed_compensation_energy_3 = specific_energy_differentiator.respond ( specific_energy);
+
+      // speed-compensation type 4 is the product of acceleration and velocity, both calculated along the heading axis
+      float3vector kalman_air_velocity;
+      kalman_air_velocity.e[NORTH] = Kalman_v_a_observer_N.get_x ( Kalman_V_A_Aoff_observer_t::VELOCITY);
+      kalman_air_velocity.e[EAST]  = Kalman_v_a_observer_E.get_x ( Kalman_V_A_Aoff_observer_t::VELOCITY);
+      kalman_air_velocity.e[DOWN]  = KalmanVario_GNSS.get_x (  	   KalmanVario_PVA_t::VARIO);
+      float air_velocity_projected = kalman_air_velocity * heading_vector;
+
+      acceleration.e[NORTH] = Kalman_v_a_observer_N.get_x ( Kalman_V_A_Aoff_observer_t::ACCELERATION);
+      acceleration.e[EAST]  = Kalman_v_a_observer_E.get_x ( Kalman_V_A_Aoff_observer_t::ACCELERATION);
+      acceleration.e[DOWN]  = KalmanVario_GNSS.get_x (	    KalmanVario_PVA_t::ACCELERATION_OBSERVED);
+      float acceleration_projected = acceleration * heading_vector;
+
+      speed_compensation_projected_4 = air_velocity_projected * acceleration_projected * RECIP_GRAVITY;
+
+#if 1 // new method under test
```

Comment:
Can this #if define being cleaned up, for better code readability?

Reply:
Basically all the other options can be removed.

Reply:
I think line 125 - 128 shall be removed then?